Zend/zend_compile.c

```diff
@@ -4884,7 +4904,15 @@ void zend_compile_implements(znode *class_node, zend_ast *ast) /* {{{ */
 }
 /* }}} */
 
-void zend_compile_class_decl(zend_ast *ast) /* {{{ */
+static zend_string *zend_generate_anon_class_name() /* {{{ */
+{
+	// TODO The opline pointer may be reused, this is not safe!
+	uint32_t next = get_next_op_number(CG(active_op_array));
+	return zend_strpprintf(0, "class@%p", &CG(active_op_array)->opcodes[next-1]);
```

Comment:
I think it is better if it uses a unique id (self-incresemenet)....will be good for debuting..